### PR TITLE
func_8007B5AC match

### PR DESCRIPTION
**Pull request by stephenjayakar**

### CODE REVIEW

File: src/melee/ft/ftcoll.c [hunk 1]
```diff
@@ -188,3 +188,54 @@ f32 func_800765F0(Fighter* fp, HSD_GObj* victim, f32 unk_floatvar) // Unk knockb
...
 }
+
+
+
+
+// 0x8007B5AC //
+// https://decomp.me/scratch/r6vQW
+
+// Permuter spit this out
+inline int inline_fn(u32 arg0)
```
Comment (ribbanya) [inline src/melee/ft/ftcoll.c:199]:
```suggestion
inline BOOL inline_fn(BOOL arg0)
```

File: src/melee/ft/ftcoll.c [hunk 2]
```diff
@@ -188,3 +188,54 @@ f32 func_800765F0(Fighter* fp, HSD_GObj* victim, f32 unk_floatvar) // Unk knockb
...
     return unk_floatvar * fp->dmg.x182c_behavior;
 }
+
+
+
+
+// 0x8007B5AC //
+// https://decomp.me/scratch/r6vQW
+
+// Permuter spit this out
```
Comment (ribbanya) [inline src/melee/ft/ftcoll.c:198]:
```suggestion
```

**Review by ribbanya — CHANGES_REQUESTED** [review]
Please run `clang-format` on your code.

**Comment by stephenjayakar** [comment]
@ribbanya ran `clang-format` on it!

File: asm/melee/ft/ftcoll.s [hunk 3]
```diff
@@ -5886,41 +5886,6 @@ lbl_8007B58C:
 /* 8007B5A4 00078184  98 07 22 1A */	stb r0, 0x221a(r7)
 /* 8007B5A8 00078188  4E 80 00 20 */	blr 
 
-.global func_8007B5AC
```
Comment (r-burns) [inline asm/melee/ft/ftcoll.s:5889]: order matters when linking, so you can't remove a function from the middle like this. it won't match

Comment (stephenjayakar) [reply asm/melee/ft/ftcoll.s:5889]: uh oh. what do I do then?

Comment (r-burns) [reply asm/melee/ft/ftcoll.s:5889]: either leave the function there and just add the C code as an example for when the rest of the functions are matched, or convert the entire ftcoll.s to inline asm in ftcoll.c and match func_8007B5AC inplace

Comment (stephenjayakar) [reply asm/melee/ft/ftcoll.s]: ok, i added the ASM back in.

Comment (stephenjayakar) [reply asm/melee/ft/ftcoll.s]: i'll keep that in mind next time

File: src/melee/ft/ftcoll.h [hunk 4]
```diff
@@ -14,6 +14,7 @@ void func_80076528(HSD_GObj* fighter_gobj); // Combo count something + adjust To
 void func_800765AC(HSD_GObj* victim); // Clear victim pointer from attacker upon freeing memory? //
 void func_800765E0(void); // Reset hitbox and phantom collision count? //
 f32 func_800765F0(Fighter* fp, HSD_GObj* victim, f32 unk_floatvar); // Unk knockback related ? //
+void func_8007B5AC(Fighter *arg0, ftHurt *arg1, s32 *arg2)
```
Comment (r-burns) [inline src/melee/ft/ftcoll.h:17]:
```suggestion
void func_8007B5AC(Fighter *arg0, ftHurt *arg1, s32 *arg2);
```

needs trailing semicolon

File: src/melee/ft/ftcoll.c [hunk 5]
```diff
@@ -188,3 +188,45 @@ f32 func_800765F0(Fighter* fp, HSD_GObj* victim, f32 unk_floatvar) // Unk knockb
...
+  R0 = arg2[7];
+  ((s32 *)(&new_var->x4_hurt1_offset.x))[3] = R7;
+  ((s32 *)(&new_var->x4_hurt1_offset.x))[4] = R0;
+  p[5] = (s32)arg2[8];
+
+  if ((inline_fn(R7) && inline_fn(R7)) && inline_fn(R7)) {
+  }
+  new_var->x1C_scale = ((f32 *)arg2)[9];
+  arg0->x221A_flag.bits.b6 = 1;
+}
```
[the remark this answers is not in the document]
Comment (r-burns) [reply src/melee/ft/ftcoll.c:232]: Maybe the missing semicolon messed up the syntax. Probably a good idea to rerun it

[the remark this answers is not in the document]
Comment (stephenjayakar) [reply src/melee/ft/ftcoll.c]: done!

File: asm/melee/ft/ftcoll.s (hunk 3 above)
Comment (ribbanya) [reply asm/melee/ft/ftcoll.s]: Currently this is just compiling the asm, and ignoring the C. You need to add both `ftcoll.c` and `ftcoll.s` to `obj_files.mk`, which might require you to port asm into the C file as-is (see other `asm` functions as examples).

**Comment by ribbanya** [comment]
Superseded by #576.

**Pull request closed without merging**